### PR TITLE
Example of OpenAPI annotations. Fixes #304

### LINKED ISSUE #304: MP Starter does not demonstrate MP Open API
The APIs in MP open api should be demonstrated. At the moment, there is no code snippet to demonstrate them. I suggest to add JAX-RS resource to demonstrate APIResponse etc.

### PR DESCRIPTION
**Pull request by rdebusscher**

### CODE REVIEW

File: src/main/resources/files/Booking.java.tpl [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+
+public class Booking {
+    @Schema(required = true, description = "Booking id")
+    private String id;
+    @Schema(required = true, description = "Description of the destination")
+    private Destination destination;
+
+    public Booking() {
```
Comment (Emily-Jiang) [inline src/main/resources/files/Booking.java.tpl:12]: Should you make this private?

File: src/main/resources/files/BookingController.java.tpl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import javax.inject.Singleton;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/booking")
+@Singleton
```
Comment (Emily-Jiang) [inline src/main/resources/files/BookingController.java.tpl:19]: I suggest to use ApplicationScoped

File: src/main/resources/files/Booking.java.tpl (hunk 1 above)
Comment (rdebusscher) [reply src/main/resources/files/Booking.java.tpl]: Removed constructor as not required

File: src/main/resources/files/BookingController.java.tpl (hunk 2 above)
Comment (rdebusscher) [reply src/main/resources/files/BookingController.java.tpl]: adjusted

**Comment by rdebusscher** [comment]
Ready for re-review

**Pull request merged**